src/libpriv/rpmostree-core.c

```diff
@@ -1107,8 +1107,14 @@ rpmostree_context_download_metadata (RpmOstreeContext *self,
           g_auto(RpmOstreeProgress) progress = { 0, };
           rpmostree_output_progress_percent_begin (&progress, "Updating metadata for '%s'",
                                                    dnf_repo_get_id (repo));
-          if (!dnf_repo_update (repo, DNF_REPO_UPDATE_FLAG_FORCE, hifstate, error))
-            return FALSE;
+          /* https://discussion.fedoraproject.org/t/rpm-ostree-giving-error/772/2
+           * This would be better fixed in libdnf but that blocks on the rebase
+           */
+          if (dnf_repo_get_kind (repo) != DNF_REPO_KIND_MEDIA)
+            {
+              if (!dnf_repo_update (repo, DNF_REPO_UPDATE_FLAG_FORCE, hifstate, error))
+                return FALSE;
```

Comment:
Were you able to test this out?
ISTM like `dnf_repo_update` would still be called on that repo when we do `dnf_context_setup_sack_with_flags()` below, no? The flow there is similar to what we have here:

```c
    ret = dnf_repo_check(repo,
                         permissible_cache_age,
                         state_local,
                         &error_local);
    if (!ret) {
        g_debug("failed to check, attempting update: %s",
                error_local->message);
        g_clear_error(&error_local);
        dnf_state_reset(state_local);
        ret = dnf_repo_update(repo,
                              DNF_REPO_UPDATE_FLAG_FORCE,
                              state_local,
                              &error_local);
```

Reply:
Ah yeah, should have noted I didn't test it beyond compilation.  Looks like you're right.